src/lib/core/command.ts

```diff
@@ -0,0 +1,23 @@
+import * as rx from "rxjs";
+import { EventEmitter } from "../../typings/embark";
+
+const COMMAND_PREFIX = "command";
+
+export abstract class Command {
+  constructor(private _name: string) {
+  }
+
+  public name() {
+    return `${COMMAND_PREFIX}:${this._name}`;
+  }
+
+  public observe(eventEmitter: EventEmitter) {
+    rx.fromEvent(eventEmitter, this.name()).subscribe(this);
```

Comment:
So one thing to keep in mind is that `.subscribe()`'ing to an observables is very similar to `.then()`'ing into a promise. Despite some characteristics where those behave very different, one thing that applies to both is that you don't want to remove the control of subscribing to consumers.

In other words, we shouldn't `.subscribe()` within a public facing API but rather return the observable so that callers can decide when they want to subscribe.

This also preserves the ability to apply operators as needed e.g.:

```
obs.observe().pipe(map(x => x)).subscribe();
```